data-generation/src/main/scala/thesis/Landy.scala

```diff
@@ -36,35 +37,74 @@ class Landy(graph: LandyAttributeGraph) extends TemporalGraph[LandyVertexPayload
    * @param interval Inclusive interval
    * @return Tuple with the activated entities
    */
-  override def activatedEntities(interval: Interval): (RDD[VertexId], RDD[EdgeId]) = throw new NotImplementedError()
+  override def activatedEntities(interval: Interval): (RDD[VertexId], RDD[EdgeId]) = {
+    val firstLocalVertices = localVertices // Remove global vertices
+      .map(vertex => (vertex._2.id, vertex._2)) // Get the global vertex ID
+      .groupByKey() // Group all
+      .map(vertex => (vertex._1, getEarliest(vertex._2.toSeq))) // Get first local vertex
+
+    val vertices = firstLocalVertices.filter(vertex =>
+      vertex._2.validFrom >= interval.start &&
+        vertex._2.validFrom <= interval.stop)
+      .map(vertex => vertex._2.id)
+
+    val edges = this.edges
+      .map(edge => (edge.attr.id, edge.attr))
+      .groupByKey()
+      .map(edge => (edge._1, getEarliest(edge._2.toSeq)))
+      .filter(edge =>
+        edge._2.validFrom >= interval.start &&
+          edge._2.validFrom <= interval.stop
+      ).map(edge => edge._1)
+    (vertices, edges)
+  }
+
+  /**
+   * @param payloads Payload updates for a single entity
+   * @return The data that corresponds to the earliest timestamp
+   */
+  private def getEarliest(payloads: Seq[LandyEntityPayload]): LandyEntityPayload = {
+    payloads.minBy(_.validFrom)
+  }
+
+  /**
+   * Get the local vertices, i.e. those who have a payload.
+   * Global vertices, which have no payload, are created when edges connect
```

Comment:
Er "Global vertices" det du har valgt å kalle disse pseudo-nodene? Som ikke egentlig skal eksistere, men må for at Graphx skal være fornøyd? Hva het de i Landy-land igjen? Instances? Entities?

Reply:
Ye, første og beste jeg kom på. Det er vel det samme som entities i Landy-land, ja. Som kan forvirres med vår "entities" som er VERTEX/EDGE